contracts/LifCrowdsale.sol

```diff
@@ -162,11 +206,16 @@ contract LifCrowdsale is Ownable, Pausable {
   }
 
   // @return true if the transaction can buy tokens
-  function validPurchase() internal constant returns (bool) {
+  function validPurchase(bool fromPresaleBuy) internal constant returns (bool) {
```

Comment:
My last ask is to separate this function into 2 separate fns: `validPurchase` and `validPresalePurchase`, they will be more focused in one task